sync15-adapter/examples/sync-pass.rs

```diff
@@ -74,27 +74,37 @@ pub struct PasswordRecord {
     pub times_used: Option<i64>,
 }
 
-fn do_auth(recur: bool) -> Result<OAuthCredentials, failure::Error> {
+fn do_auth(cfg: Config) -> Result<FirefoxAccount, failure::Error> {
     match fs::File::open("./credentials.json") {
         Err(_) => {
-            if recur {
-                panic!("Failed to open credentials 2nd time");
-            }
-            println!("No credentials found, invoking boxlocker.py...");
-            process::Command::new("python")
-                .arg("../boxlocker/boxlocker.py").output()
-                .expect("Failed to run boxlocker.py");
-            return do_auth(true);
+            println!("No credentials found, so let's do an OAuth flow.");
+            let mut acct = FirefoxAccount::new(cfg, CLIENT_ID, REDIRECT_URI);
+            let oauth_uri = acct.begin_oauth_flow(&[SYNC_SCOPE], true)?;
+            println!("Please visit this URL, sign in, and then copy-paste the final URL below.");
+            println!("");
+            println!("    {}", oauth_uri);
+            println!("");
+            let final_url = url::Url::parse(&prompt_string("Final URL").unwrap())?;
+            let mut code = String::new();
+            let mut state = String::new();
+            for param in final_url.query_pairs() {
+              match param {
+                (Cow::Borrowed("code"), c) => { code = c.into_owned() },
+                (Cow::Borrowed("state"), s) => { state = s.into_owned() },
+                _ => {}
+              }
+            };
+            acct.complete_oauth_flow(&code, &state)?;
+            let mut file = fs::File::create("./credentials.json")?;
+            write!(file, "{}", acct.to_json()?)?;
+            file.flush()?;
+            Ok(acct)
         },
         Ok(mut file) => {
             let mut s = String::new();
             file.read_to_string(&mut s)?;
-            let creds: OAuthCredentials = serde_json::from_str(&s)?;
-            let time = SystemTime::now().duration_since(UNIX_EPOCH)?.as_secs();
-            if creds.expires_in + creds.auth_at < time {
-                println!("Warning, credentials may be stale.");
-            }
-            Ok(creds)
+            let acct = FirefoxAccount::from_json(&s)?;
```

Comment:
I assume this crashes and burns on existing files (which don't store data in the same format)? Maybe add a warning if that call fails, that you need to delete credentials.json, since otherwise the error returned will likely be quite opaque (admittedly, the number of people migrating is fairly low, but i don't think this should be that much work either).

Reply:
I added some logic to detect this case and re-do the oauth flow.